pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -0,0 +1,86 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSXServiceAccountSpec defines the desired state of NSXServiceAccount
+type NSXServiceAccountSpec struct {
+	VPCName string `json:"vpcName,omitempty"`
+}
+
+type NSXProxyEndpointAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	//+kubebuilder:validation:Format=ip
+	IP string `json:"ip,omitempty"`
+}
+
+type NSXProxyProtocol string
+
+const (
+	NSXProxyProtocolTCP NSXProxyProtocol = "TCP"
+)
+
+type NSXProxyEndpointPort struct {
+	Name     string           `json:"name,omitempty"`
+	Port     uint16           `json:"port,omitempty"`
+	Protocol NSXProxyProtocol `json:"protocol,omitempty"`
+}
+
+type NSXProxyEndpoint struct {
+	Addresses []NSXProxyEndpointAddress `json:"addresses,omitempty"`
+	Ports     []NSXProxyEndpointPort    `json:"ports,omitempty"`
+}
+
+type NSXSecret struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type NSXServiceAccountPhase string
+
+const (
+	NSXServiceAccountPhaseRealized   NSXServiceAccountPhase = "realized"
+	NSXServiceAccountPhaseInProgress NSXServiceAccountPhase = "inProgress"
```

Comment:
I didn't find where NSXServiceAccountPhaseInProgress is set. Can we set it if deletionTimestamp=0 and phase!=NSXServiceAccountPhase? Or if the time for realizing NSXServiceAccount is really short, we can delete this NSXServiceAccountPhaseInProgress.

Reply:
do we still need NSXServiceAccountPhaseInProgress?

Reply:
Current not used, but this commit is merged to main branch. We can ignore CRD definition part in this PR.

Reply:
Cherry-picked CRD PRs from main to vpc_dev branch, this part is no longer in this PR.